built-in-services/account/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+        }
+
+        // multi sig
+        if wit.signature_type != ACCOUNT_TYPE_MULTI_SIG {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                115,
+                "signature_type not valid".to_owned(),
+            );
+        }
+
+        if wit.signatures.len() != wit.pubkeys.len() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                116,
+                "len of signatures and pubkeys must be equal".to_owned(),
+            );
+        }
+
+        if wit.signatures.len() > MAX_PERMISSION_ACCOUNTS as usize {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                117,
+                "len of signatures must be [1,16]".to_owned(),
+            );
+        }
+
+        let permission_res: Option<Permission> = self.sdk.get_account_value(&wit.sender, &0u8);
+
+        if permission_res.is_none() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                117,
+                "account not existed".to_owned(),
+            );
+        }
+
+        let permission = permission_res.unwrap();
+        let mut weight_sum = 0;
+        let size = permission.accounts.len();
+        let mut hash_account = [false; MAX_PERMISSION_ACCOUNTS as usize];
+
+        for i in 0..wit.signatures.len() {
+            let res = verify_single_sig(&payload.tx_hash, &wit.signatures[i], &wit.pubkeys[i]);
+            if res.is_error() {
+                continue;
+            }
+
+            for (k, item) in permission.accounts.iter().enumerate().take(size) {
+                if hash_account[k] {
+                    continue;
+                }
+                if item.address.eq(&res.succeed_data.address) {
+                    hash_account[k] = true;
+                    weight_sum += item.weight;
+                    break;
+                }
+            }
+
+            if weight_sum >= permission.threshold {
+                return ServiceResponse::<VerifyResponse>::from_succeed(VerifyResponse {
+                    address: wit.sender,
+                });
+            }
+        }
+
+        ServiceResponse::<VerifyResponse>::from_error(
+            111,
+            "multi signature not verified".to_owned(),
+        )
+    }
+
+    #[cycles(100_00)]
+    #[read]
+    fn get_account_from_address(
+        &self,
+        ctx: ServiceContext,
+        payload: GetAccountPayload,
+    ) -> ServiceResponse<GenerateAccountResponse> {
+        let permission = self
+            .sdk
+            .get_account_value(&payload.user, &0u8)
+            .unwrap_or(Permission {
+                accounts:  Vec::<Account>::new(),
+                threshold: 0,
+            });
+
+        if permission.threshold == 0 {
+            return ServiceResponse::<GenerateAccountResponse>::from_error(
+                110,
+                "account not existed".to_owned(),
+            );
+        }
+
+        let mut accounts = Vec::<PayloadAccount>::new();
```

Comment:
It seems that variable `accounts` is not related to the method response ?

Reply:
right, response is modified,i  forgot to delete the accounts